### PR TITLE
Feature/spawn points

### PR DESCRIPTION
**Pull request by NDobrev**

### CODE REVIEW

File: Games/GiftOfTheSanctum/Source/GiftOfTheSanctumGame.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+	for (uint8_t playerIndex = 0u; playerIndex < MaxPlayers; playerIndex++)
+	{
+		InitializePlayerController(playerIndex);
+	}
+
+	// Gather spawn points
+	Zmey::tmp::vector<Zmey::EntityId> spawnPoints = tagManager.FindAllByTag(Zmey::Name("SpawnPoint"));
+	m_SpawnPoints = EntityVector(spawnPoints.begin(), spawnPoints.end());
+	// Setup player on swapn points
```
Comment (Alekssasho) [inline Games/GiftOfTheSanctum/Source/GiftOfTheSanctumGame.cpp]: spawn

File: Games/GiftOfTheSanctum/Source/GiftOfTheSanctumGame.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <Zmey/EngineLoop.h>
+#include <Zmey/Game.h>
+#include <Zmey/Utilities.h>
+#include <Zmey/EntityManager.h>
+#include <Zmey/Math/Math.h>
+
+using EntityVector = std::vector<Zmey::EntityId>;
```
Comment (Alekssasho) [inline Games/GiftOfTheSanctum/Source/GiftOfTheSanctumGame.h]: Zmey::stl::vector, as this is not using the right allocator

File: Games/GiftOfTheSanctum/zmeyconfig.ini [hunk 3]
```diff
@@ -1,4 +1,5 @@
 [InputController]
-Actions=WalkX,CONT+LeftStickX
-Actions=WalkZ,CONT+LeftStickY
-Actions=Cast,FaceDown
+Actions=WalkX0,CONT+LeftStickX
```
Comment (Alekssasho) [inline Games/GiftOfTheSanctum/zmeyconfig.ini]: what is this 0 ?

File: Projects/Zmey/Zmey.vcxproj [hunk 4]
```diff
@@ -75,7 +75,7 @@
     <Link>
       <AdditionalLibraryDirectories>$(SolutionDir)ThirdParty\libx64;$(SolutionDir)ThirdParty\libx64\$(Configuration);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
       <AdditionalDependencies>PxPvdSDK_x64.lib;PhysX3Extensions.lib;PhysX3_x64.lib;PxFoundation_x64.lib;vulkan-1.lib;assimp-vc140-mt.lib;d3d12.lib;dxgi.lib;Xinput.lib;easy_profiler.lib;%(AdditionalDependencies)</AdditionalDependencies>
-      <TreatLinkerWarningAsErrors>true</TreatLinkerWarningAsErrors>
+      <TreatLinkerWarningAsErrors>false</TreatLinkerWarningAsErrors>
```
Comment (Alekssasho) [inline Projects/Zmey/Zmey.vcxproj]: revert

File: Source/Components/MeshComponentManager.cpp [hunk 5]
```diff
@@ -53,8 +53,8 @@ void MeshComponentManager::InitializeFromBlob(const tmp::vector<EntityId>& entit
 		stream >> name;
 		ASSERT(Zmey::Modules::ResourceLoader->IsResourceReady(name));
 		Graphics::MeshHandle meshHandle = *Zmey::Modules::ResourceLoader->AsMeshHandle(name);
-		m_EntityToIndex[entities[i]] = uint32_t(m_Meshes.size());
 		m_Meshes.push_back(meshHandle);
+		m_EntityToIndex[entities[i]] = uint32_t(m_Meshes.size()) - 1;
```
Comment (Alekssasho) [inline Source/Components/MeshComponentManager.cpp]: revert

File: Tools/BlenderPlugins/scripts/addons/zmey_properties/__init__.py [hunk 6]
```diff
@@ -108,7 +108,8 @@ def execute(self, context):
                         "glTF_node_index" : current_index
                     }
                     # check for color inside the material
-                    if len(obj.data.materials) > 0 and obj.data.materials[0].diffuse_color:
+                    print(hasattr(obj.data, 'materials'));
```
Comment (Alekssasho) [inline Tools/BlenderPlugins/scripts/addons/zmey_properties/__init__.py]: remove

File: Tools/BlenderPlugins/scripts/addons/zmey_properties/__init__.py [hunk 7]
```diff
@@ -108,7 +108,8 @@ def execute(self, context):
                         "glTF_node_index" : current_index
                     }
                     # check for color inside the material
-                    if len(obj.data.materials) > 0 and obj.data.materials[0].diffuse_color:
+                    print(hasattr(obj.data, 'materials'));
+                    if  hasattr(obj.data, 'materials') and len(obj.data.materials) > 0 and obj.data.materials[0].diffuse_color:
```
Comment (Alekssasho) [inline Tools/BlenderPlugins/scripts/addons/zmey_properties/__init__.py]: if obj.data.materials and ....

File: Games/GiftOfTheSanctum/Source/GiftOfTheSanctumGame.cpp (hunk 1 above)
Comment (NDobrev) [reply Games/GiftOfTheSanctum/Source/GiftOfTheSanctumGame.cpp]: fixed

File: Games/GiftOfTheSanctum/Source/GiftOfTheSanctumGame.h (hunk 2 above)
Comment (NDobrev) [reply Games/GiftOfTheSanctum/Source/GiftOfTheSanctumGame.h]: added todo

File: Games/GiftOfTheSanctum/zmeyconfig.ini (hunk 3 above)
Comment (NDobrev) [reply Games/GiftOfTheSanctum/zmeyconfig.ini]: reverted

File: Tools/BlenderPlugins/scripts/addons/zmey_properties/__init__.py (hunk 6 above)
Comment (NDobrev) [reply Tools/BlenderPlugins/scripts/addons/zmey_properties/__init__.py]: done

File: Tools/BlenderPlugins/scripts/addons/zmey_properties/__init__.py (hunk 7 above)
Comment (NDobrev) [reply Tools/BlenderPlugins/scripts/addons/zmey_properties/__init__.py]: done

File: Source/Components/MeshComponentManager.cpp (hunk 5 above)
Comment (NDobrev) [reply Source/Components/MeshComponentManager.cpp]: done

File: Projects/Zmey/Zmey.vcxproj (hunk 4 above)
Comment (NDobrev) [reply Projects/Zmey/Zmey.vcxproj]: fixed

File: Tools/BlenderPlugins/scripts/addons/zmey_properties/__init__.py [hunk 8]
```diff
@@ -108,7 +108,7 @@ def execute(self, context):
                         "glTF_node_index" : current_index
                     }
                     # check for color inside the material
-                    if len(obj.data.materials) > 0 and obj.data.materials[0].diffuse_color:
+                    if  obj.materials and len(obj.data.materials) > 0 and obj.data.materials[0].diffuse_color:
```
Comment (Alekssasho) [inline Tools/BlenderPlugins/scripts/addons/zmey_properties/__init__.py]: obj.data.materials

**Pull request merged**